Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,876 @@
+                                                     LogicOperator.Or))
+
+        self.declareProperty(MatrixWorkspaceProperty('BeamInputWorkspace', '',
+                                                     direction=Direction.Input,
+                                                     optional=PropertyMode.Optional),
+                             doc='The name of the empty beam input workspace.')
+
+        self.setPropertySettings('BeamInputWorkspace', transmission)
+
+        self.declareProperty(MatrixWorkspaceProperty('CadmiumTransmissionInputWorkspace', '',
+                                                     direction=Direction.Input,
+                                                     optional=PropertyMode.Optional),
+                             doc='The name of the cadmium transmission input workspace.')
+
+        self.setPropertySettings('CadmiumTransmissionInputWorkspace', EnabledWhenProperty(transmission, beam, LogicOperator.Or))
+
+        self.declareProperty(MatrixWorkspaceProperty('TransmissionInputWorkspace', '',
+                                                     direction=Direction.Input,
+                                                     optional=PropertyMode.Optional),
+                             doc='The name of the transmission input workspace.')
+
+        self.setPropertySettings('TransmissionInputWorkspace', reduction)
+
+        self.declareProperty(WorkspaceGroupProperty('ContainerInputWorkspace', '',
+                                                    direction=Direction.Input,
+                                                    optional=PropertyMode.Optional),
+                             doc='The name of the container workspace.')
+
+        self.setPropertySettings('ContainerInputWorkspace', reduction)
+
+        self.declareProperty(WorkspaceGroupProperty('QuartzInputWorkspace', '',
+                                                    direction=Direction.Input,
+                                                    optional=PropertyMode.Optional),
+                             doc='The name of the polarisation efficiency correction workspace.')
+
+        self.setPropertySettings('QuartzInputWorkspace',
+                                 EnabledWhenProperty(vanadium, sample, LogicOperator.Or))
+
+        self.declareProperty(WorkspaceGroupProperty('VanadiumInputWorkspace', '',
+                                                    direction=Direction.Input,
+                                                    optional=PropertyMode.Optional),
+                             doc='The name of the vanadium workspace.')
+
+        self.setPropertySettings('VanadiumInputWorkspace', EnabledWhenProperty('DetectorEfficiencyCalibration',
+                                                                               PropertyCriterion.IsEqualTo, 'Vanadium'))
+
+        self.declareProperty('AverageScan', False,
```

Comment:
Instead of declaring two booleans, it is preferred to declare one string property where you can choose among: `OutputIndividualScans`, `AverageScans`, `SumScans`